main.tf

```diff
@@ -98,3 +98,11 @@ resource "aws_s3_bucket_server_side_encryption_configuration" "main" {
     }
   }
 }
+
+
+resource "aws_s3_bucket_logging" "main" {
+  count         = length(var.s3_logging) > 0 ? 1 : 0
+  bucket        = aws_s3_bucket.main.id
+  target_bucket = var.s3_logging["target_bucket"]
+  target_prefix = var.s3_logging["target_prefix"]
+}
```

Comment:
There are some missing features in this block. Please include those
Ref: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_logging

@ndegwajohn